crates/ruff/src/rules/flake8_implicit_str_concat/rules/implicit.rs

```diff
@@ -103,24 +106,39 @@ pub(crate) fn implicit(
         })
         .tuple_windows()
     {
-        if a_tok.is_string() && b_tok.is_string() {
-            if locator.contains_line_break(TextRange::new(a_range.end(), b_range.start())) {
-                diagnostics.push(Diagnostic::new(
-                    MultiLineImplicitStringConcatenation,
-                    TextRange::new(a_range.start(), b_range.end()),
-                ));
-            } else {
-                let mut diagnostic = Diagnostic::new(
-                    SingleLineImplicitStringConcatenation,
-                    TextRange::new(a_range.start(), b_range.end()),
-                );
-
-                if let Some(fix) = concatenate_strings(*a_range, *b_range, locator) {
-                    diagnostic.set_fix(fix);
-                }
-
-                diagnostics.push(diagnostic);
-            };
+        let (a_range, b_range) = match (a_tok, b_tok) {
+            (Tok::String { .. }, Tok::String { .. }) => (*a_range, *b_range),
+            (Tok::String { .. }, Tok::FStringStart) => (
+                *a_range,
+                indexer.fstring_ranges().innermost(b_range.start()).unwrap(),
```

Comment:
I find the use of `fstring_ranges` here a bit unfortunate because:

* Looking up the range is at least: `O(log(n))`
* So many unwraps

But I guess it's fine, considering that implicit concatenated strings are somewhat rare

Reply:
Yes, I agree. This could be solved with the introduction of a new `StringList` node though as the AST would contain the range, so no lookups and no unwraps 😉